### PR TITLE
Update provider pinning, add context.tf, and add auto-release

### PR DESCRIPTION
**Pull request by joe-niland**
## what
* Relax version constraints for TF and providers
* Update module to use context.tf pattern
* Add auto-release Action

## why
* Allow TF 0.13+, AWS provider 3+, etc
* Streamline common module variable usage
* Allow auto-release on merge to master

## references
* None

### CODE REVIEW

**Review by Gowiem — DISMISSED** [review]
Looks solid :shipit:

File: test/src/Makefile [hunk 1]
```diff
@@ -7,6 +7,7 @@ PATH 	:= $(PATH):$(GOBIN)
```
Comment (aknysh) [inline test/src/Makefile]: please update this Makefile, use the content from https://github.com/cloudposse/terraform-aws-service-control-policies/blob/master/test/src/Makefile

**Review by aknysh — COMMENTED** [review]
please see comments

**Comment by aknysh** [comment]
@joe-niland 
we need to update to `go` modules

**Comment by aknysh** [comment]
remove `Gopkg.lock` and `Gopkg.toml`

**Comment by aknysh** [comment]
then execute these commands:

```
cd test/src/
go mod init github.com/cloudposse/$(cd ../../ && basename `pwd`)
go mod tidy

```

**Review by aknysh — APPROVED** [review]
please see comments

**Pull request merged**